src/host/coreaudio/macos/device.rs

```
@@ -392,6 +403,37 @@ impl Device {
         })
     }
 
+    fn id(&self) -> Result<DeviceId, DeviceIdError> {
+        let property_address = AudioObjectPropertyAddress {
+            mSelector: kAudioDevicePropertyDeviceUID,
+            mScope: kAudioObjectPropertyScopeGlobal,
+            mElement: kAudioObjectPropertyElementMain,
+        };
+        let mut name: CFStringRef = std::ptr::null_mut();
+        let data_size = size_of::<CFStringRef>() as u32;
+        let status = unsafe {
+            AudioObjectGetPropertyData(
+                self.audio_device_id,
+                NonNull::from(&property_address),
+                0,
+                null(),
+                NonNull::from(&data_size),
+                NonNull::from(&mut name).cast(),
+            )
+        };
+        if status == 0 {
```

Instead of checking `status == 0` would it make sense to `check_os_status(status)?` consistent with the rest of the code?

Then instead (or additionally) we should return an error `if uid.is_null()`.